Documentation/contributor-guide/features.md

```diff
@@ -0,0 +1,60 @@
+# Features 
+
+This document provides an overview of etcd features and general guidelines for adding and removing features. The project maintainers can override the guidelines per the need of the project following the project governance.
+
+The etcd features fall into three stages, experimental, stable, and unsafe.
+
+## Experimental
+
+Any new feature should be added as an experimental feature. An experimental feature is characterized as below:
+- Might be buggy due to lack of user testing. Enabling the feature may not work as expected.
+- Support for such a feature may be dropped at any time without notice
+- Disabled by default when added initially
+
+### Minimum development requirements
+
+Experimental feature related development work is expected to be high quality, similar to any development work for the project. A related PR should meet the following criteria,
+- Associated with an issue providing a clear need for such a feature
+- Any configuration flags related to the implementation of the feature are prefixed with `experimental` e.g. `--experimental-feature-name`
+- Any variable names related to the implementation of the feature are prefixed with `Experimental` e.g. `ExperimentalFeatureName`
+- Provided with unit tests
+- Provided with e2e test coverage for at least the basic scenario
+- Added a CHANGELOG entry
+- Disabled by default
+- Approved by at least two project maintainers
+
+### Graduation to Stable
+
+It is important that experimental features don't get stuck in that stage. After a release of an intiial enablement (e.g. etcd v3.5), an experimental feature should be revisited in the following release (e.g. etcd v3.6 or after six months if release takes longer) and see if it's ready for graduation to the stable stage in the future release (e.g. etcd v3.7 or v4.0). If it's ready per discussions with the project maintainers, then depending on the scope of the feature, at least the following enhancements should be consiered to be worked on by opening a PR(s) and new issue(s) for tracking as needed,
+- Fix any open issues against the feature
+- Add robust e2e tests
+- Enhance logs for proper debugging
+- Update metrics and benchmarks if needed
```

Comment:
Maybe robust tests, logs, metrics and benchmarks should be part of minimum dev requirements. Otherwise this creates an expectation that somebody will pick up this work during 'graduation' phase. This will slow down graduation process. Also original author probably has the best context to add those tests/logs.

For example `experimental-compaction-batch-limit` https://github.com/oss-study/etcd-3.4.9/commit/1e213b7ab6991d67722e7e1496a93b3a7fae0366#diff-0abf7361537ed66a5b8d85cdc518a4ab2f693a1e8c929c93de3746851a5ddfbe

Technically it's just exposing hardcoded value. But to graduate it, do we need more tests/benchmarks?